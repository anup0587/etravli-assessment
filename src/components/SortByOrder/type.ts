export interface SortByOrderProps {
  optionList: Array<{ label: string; value: string }>;
  onClickItem:(_value:string)=>void;
  selectedValue:string;
}
