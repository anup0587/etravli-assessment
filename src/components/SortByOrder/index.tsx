import { Box, Button, Menu, MenuItem } from "@mui/material";
import React, { useState } from "react";
import { SortByOrderProps } from "./type";

const SortByOrder = ({
  optionList,
  onClickItem,
  selectedValue,
}: SortByOrderProps) => {
  
  const [anchorEl, setAnchorEl] = useState(null);

  const selectedIndex =
    optionList && optionList.findIndex((item) => item.value === selectedValue);

  const handleClick = (event: { currentTarget: any }) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const handleClickItem = (value: string) => {
    onClickItem(value);
    setAnchorEl(null);
  };
  return (
    <Box>
      <Button
        aria-controls="simple-menu"
        aria-haspopup="true"
        variant="outlined"
        onClick={handleClick}
      >
        Sort by
      </Button>
      <Menu
        id="simple-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        {optionList &&
          optionList.map((option: any, index: number) => (
            <MenuItem
              key={option.value}
              selected={index === selectedIndex}
              onClick={() => handleClickItem(option.value)}
            >
              {option.label}
            </MenuItem>
          ))}
      </Menu>
    </Box>
  );
};

export default SortByOrder;
