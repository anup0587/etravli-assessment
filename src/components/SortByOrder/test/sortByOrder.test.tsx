import { render } from "@testing-library/react";
import SortByOrder from "..";

describe("SortByOrder", () => {
  const props = {
    optionList: [
      { label: "Label 1", value: "value1" },
      { label: "Label 2", value: "value2" },
    ],

    selectedValue: "value1",
  };
  it(" snapshot", () => {
    const { container } = render(
      <SortByOrder {...props} onClickItem={() => {}} />
    );
    expect(container).toMatchSnapshot();
  });
});
