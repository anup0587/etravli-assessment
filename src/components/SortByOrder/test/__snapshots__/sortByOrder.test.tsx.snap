// Vitest Snapshot v1

exports[`SortByOrder >  snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <button
      aria-controls="simple-menu"
      aria-haspopup="true"
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Sort by
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
